test/e2e/common.go

```diff
@@ -308,6 +308,33 @@ func ensureControlPlaneInitialized(ctx context.Context, input clusterctl.ApplyCu
 	result.ControlPlane = controlPlane
 }
 
+// ensureContolPlaneReplicasMatch waits for the control plane machine replicas to be created.
+func ensureContolPlaneReplicasMatch(ctx context.Context, proxy framework.ClusterProxy, ns, clusterName string, replicas int, intervals []interface{}) {
+	By("Waiting for all control plane nodes to exist")
+	inClustersNamespaceListOption := client.InNamespace(ns)
+	// ControlPlane labels
+	matchClusterListOption := client.MatchingLabels{
+		clusterv1.MachineControlPlaneLabel: "",
+		clusterv1.ClusterNameLabel:         clusterName,
+	}
+
+	Eventually(func() (int, error) {
+		machineList := &clusterv1.MachineList{}
+		lister := proxy.GetClient()
+		if err := lister.List(ctx, machineList, inClustersNamespaceListOption, matchClusterListOption); err != nil {
+			Logf("Failed to list the machines: %+v", err)
+			return 0, err
+		}
+		count := 0
+		for _, machine := range machineList.Items {
+			if machine.Status.NodeRef != nil {
```

Comment:
Is it possible to have a more granular check here instead of checking for _not nil_ ?